### DISCUSSION TITLE
Use of new SVG icon pack?

### DISCUSSION THREAD

**Discussion by The-Lum** [plantuml/plantuml#965] [Ideas]
Hello PlantUML team, and all,

Thanks a lot for:
- Use of [OpenIconic](https://plantuml.com/openiconic)
- #820

And similar to:
- https://github.com/plantuml/plantuml/discussions/820#discussioncomment-1950436
- #629
- #141
- https://github.com/plantuml/plantuml/issues/119#issuecomment-1068976607

_(even if the road map is already well filled...)_

- **Is it plan to add more icon or icon library?**
_For example, especially, in order to have a SVG icon of database or other..._

---

_Hints:_
- iconoir: https://iconoir.com / https://github.com/lucaburgio/iconoir
- openmoji: https://openmoji.org / https://github.com/hfg-gmuend/openmoji
- _or other..._

---

_To debate..._
Regards.

**Comment by Potherca** [comment]
I started the https://github.com/plantuml-stdlib/ organisation in an attempt to bring various different icon/sprite libs together so duplicate effort could be avoided. (Such as each repo having its own logic to generate sprites from SVGs).

Part of the long-term aim was to make it trivial to add _new_ icon/sprite sets, optionally removing the need for separate projects _entirely_. (Or at the very least for new sets).

The suggestion to add Iconoir and/or Openmoji looks like a great opportunity for a first proof-of-concept for this.  

@The-Lum Is this something you would like to collaborate on?

**Reply by The-Lum** [reply]
>Is this something you would like to collaborate on?

I saw it easier.
And accordingly: 
- #1066

I don't turn all SVGs into SVG sprites; and I wouldn't want to overload the weight of the final diagram...
But rather, like the emoji, find them in relation to a directory structure...

_To debate how to achieve that..._

**Reply by Potherca** [reply]
Currently, within the stdlib repos, there are many different ways things are done. Some repos only make one file with all sprites, others have one file per sprite but all in one folder, others use separate folders. As almost all repos have some sort of automated script for their process, it should be possible to move towards one sane way to do it for all sprite projects under stdlib.

> To debate how to achieve that...

Very true. It'll take some thinking about and some discussions to come to a good solution.
